python-frontend/src/test/java/org/sonar/python/semantic/ProjectLevelSymbolTableTest.java

```diff
@@ -198,4 +201,78 @@ public void aliased_imported_symbols() {
+    );
+    Symbol importedASymbol = tree.globalVariables().iterator().next();
+    assertThat(importedASymbol.name()).isEqualTo("A");
+    assertThat(importedASymbol.fullyQualifiedName()).isEqualTo("mod1.A");
+    assertThat(importedASymbol.kind()).isEqualTo(Symbol.Kind.CLASS);
+    ClassSymbol classA = (ClassSymbol) importedASymbol;
+    assertThat(classA.hasUnresolvedTypeHierarchy()).isEqualTo(false);
+    assertThat(classA.superClasses()).hasSize(1);
+    assertThat(classA.superClasses().get(0).kind()).isEqualTo(Symbol.Kind.CLASS);
+
+    tree = parse(
+      new SymbolTableBuilder("my_package", pythonFile("my_module.py"), globalSymbols),
+      "import mod1",
+      "mod1.A"
+    );
+
+    QualifiedExpression qualifiedExpression = PythonTestUtils.getFirstChild(tree, t -> t.is(Tree.Kind.QUALIFIED_EXPR));
+    importedASymbol = qualifiedExpression.name().symbol();
+    assertThat(importedASymbol.name()).isEqualTo("A");
+    assertThat(importedASymbol.fullyQualifiedName()).isEqualTo("mod1.A");
+    assertThat(importedASymbol.kind()).isEqualTo(Symbol.Kind.CLASS);
+    classA = (ClassSymbol) importedASymbol;
+    assertThat(classA.hasUnresolvedTypeHierarchy()).isEqualTo(false);
+    assertThat(classA.superClasses()).hasSize(1);
+    assertThat(classA.superClasses().get(0).kind()).isEqualTo(Symbol.Kind.CLASS);
+  }
+
+  @Test
+  public void not_class_symbol_in_super_class() {
+    ClassSymbolImpl classASymbol = new ClassSymbolImpl("A", "mod1.A");
+    classASymbol.addSuperClass(new SymbolImpl("foo", "mod1.foo"));
+    FileInput tree = parse(
+      new SymbolTableBuilder("my_package", pythonFile("my_module.py"), Collections.singletonMap("mod1", Collections.singleton(classASymbol))),
+      "from mod1 import A"
+    );
+
+    Symbol importedASymbol = tree.globalVariables().iterator().next();
+    assertThat(importedASymbol.kind()).isEqualTo(Symbol.Kind.CLASS);
+    ClassSymbol classA = (ClassSymbol) importedASymbol;
+    assertThat(classA.hasUnresolvedTypeHierarchy()).isEqualTo(true);
+    assertThat(classA.superClasses()).hasSize(1);
+    assertThat(classA.superClasses().get(0).kind()).isEqualTo(Symbol.Kind.OTHER);
+  }
+
+  @Test
+  public void builtin_symbol_in_super_class() {
+    ClassSymbolImpl classASymbol = new ClassSymbolImpl("A", "mod1.A");
+    classASymbol.addSuperClass(new SymbolImpl("BaseException", "BaseException"));
+    FileInput tree = parse(
+      new SymbolTableBuilder("my_package", pythonFile("my_module.py"), Collections.singletonMap("mod1", Collections.singleton(classASymbol))),
+      "from mod1 import A"
+    );
+
+    Symbol importedASymbol = tree.globalVariables().iterator().next();
+    assertThat(importedASymbol.kind()).isEqualTo(Symbol.Kind.CLASS);
+    ClassSymbol classA = (ClassSymbol) importedASymbol;
+    assertThat(classA.hasUnresolvedTypeHierarchy()).isEqualTo(false);
+    assertThat(classA.superClasses()).hasSize(1);
+    assertThat(classA.superClasses().get(0).kind()).isEqualTo(Symbol.Kind.OTHER);
```

Comment:
I have the feeling that the following 2 things contradict each other:
* having a superclass which is not a class symbol
* `hasUnresolvedTypeHierarchy()` returning `false`